examiner-baseline.ipynb

```diff
@@ -0,0 +1 @@
uffle=True,num_workers=4)\nval_dl  = DataLoader(val_ds,batch_size=BATCH_SIZE,shuffle=True,num_workers=4)","metadata":{"trusted":true},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"import torch.optim","metadata":{"trusted":true},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"opt = torch.optim.AdamW([*model.parameters(),*mlp.parameters()],lr=5e-4,weight_decay=5e-4)","metadata":{"trusted":true},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"device = \"cuda\"\n\nmodel.to(device)\nmlp.to(device)\n\n\nfor epoch in range(500):\n    model.train()\n    mlp.train()\n    tr_loss = []\n    res = []\n    for step, batch in enumerate(train_dl):\n        imgs,labels = batch\n        encodings = model(imgs.to(device))\n        logits = mlp(encodings.pooler_output.flatten(1))\n        loss = nn.functional.cross_entropy(logits,labels.to(device))\n        tr_loss.append(loss.detach().cpu())\n        res.append(torch.argmax(logits,dim=1).detach()==labels.to(device))\n        loss.backward()\n        opt.step()\n        opt.zero_grad()\n    tr_loss = torch.stack(tr_loss).float().mean() \n    tr_acc = torch.cat([i for i in res]).float().mean()\n    model.eval()\n    mlp.eval()\n    res = []\n    val_loss = []\n    for step, batch in enumerate(val_dl):\n        with torch.no_grad():\n            imgs,labels = batch\n            encodings = model(imgs.to(device))\n            logits = mlp(encodings.pooler_output.flatten(1))\n            loss = nn.functional.cross_entropy(logits,labels.to(device))\n            res.append(torch.argmax(logits,dim=1)==labels.to(device))\n            val_loss.append(loss.detach())\n    print(f\"{epoch} Tr loss: {tr_loss} Tr acc: {tr_acc}\",\"Val loss: \",torch.stack(val_loss).float().mean(),\"Val acc: \",torch.cat([i for i in res]).float().mean())\n        ","metadata":{"trusted":true},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"","metadata":{},"execution_count":null,"outputs":[]}]}
```

Comment:
What is this notebook for?

Reply:
This is the notebook exported from Kaggle. It does not really matter, and it is bascially the same as the .py file.